packages/@sanity/base/src/components/react-track-elements/createScope.tsx

```diff
@@ -2,24 +2,53 @@ import * as React from 'react'
 import {TrackerContext} from './types'
 import {createUseReporter, IsEqualFunction} from './createUseReporter'
 import {createStore} from './createStore'
+import {Reported} from './index'
 
-const useReporterGuard = (id: string) => {
-  throw new Error(
-    `No context provided for reporter. Make sure that the component calling "useReporter(${id}, ...)", is wrapped in a <Tracker> element`
-  )
+let didWarn = false
```

Comment:
I'm sure this isn't a problem at all, just noting some thoughts:

Are we sure that the value isn't initialized more than once in some weird way since it's put there in the outer scope? If more than one thing imports this file?

We could move the initialization into the function we're memoizing to have complete control over when it's initialized :)

Reply:
The script is only executed once even if it's imported/require'd several times, so can't really see how that can happen. That said, it's now a bit random what error you'll see, so memoizing the individual functions and show errors for all of them could be a worthwhile improvement. Let's make a note of that and revisit later?